src/payment/strategies/laybuy/laybuy-payment-strategy.ts

```diff
@@ -0,0 +1,79 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+import { some } from 'lodash';
+
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { RequestError } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+export default class LaybuyPaymentStrategy implements PaymentStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _formPoster: FormPoster
+    ) {}
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentData = payment && payment.paymentData;
+
+        if (!payment || !paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({ ...payment, paymentData }))
+            )
+            .catch(error => {
+                if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'insert error here' })) {
+                    return new Promise(() => {
+                        this._formPoster.postForm('https://sandbox-api.laybuy.com/order/create', {
+                            headers: {
+                                Accept: 'application/json',
+                                'Content-Type': 'application/json',
+                                authorization: 'Basic cGF5bWVudC1pbnRlZ3JhdGlvbnMrbnpAYmlnY29tbWVyY2UuY29tOk9UTXlOME09',
+                            },
+                        });
+                    });
+                }
```

Comment:
At the end when BigPay code has been finished, this code will see something like this:
```suggestion
                    return new Promise(() => {
                        this._formPoster.postForm(**Url from errors.body.provider_data.redirect_url**, { });
                    });
                }
```